packages/bytecode-utils/src/lib/bytecode.ts

```diff
@@ -0,0 +1,113 @@
+import { arrayify, hexlify } from '@ethersproject/bytes';
+import bs58 from 'bs58';
+import * as CBOR from 'cbor-x';
+
+type CBOR = {
+  bytes: string;
+  length: number;
+};
+
+// eslint-disable-next-line functional/no-mixed-type
+type DecodedObject = {
+  ipfs?: string;
+  solcVersion?: string;
+  [key: string]: string | Uint8Array | undefined | boolean;
+};
+
+/**
+ * Decode contract's bytecode
+ * @param bytecode - hex of the bytecode with 0x prefix
+ * @returns Object describing the contract
+ */
+export const decode = (bytecode: string): DecodedObject => {
+  if (bytecode.length === 0) {
+    throw Error('Bytecode cannot be null');
+  }
+  if (bytecode.substring(0, 2) !== '0x') {
+    throw Error('Bytecode should start with 0x');
```

Comment:
Also here, why not just handle bytecodes without 0x?
```suggestion
    bytecode = "0x" + bytecode
```